lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -843,6 +875,105 @@ void FunctionEntrypointAnalyzer::interproceduralPropagation() {
   }
 }
 
+void FEA::initializeMapForDeductions(FunctionSummary &Summary,
+                                     abi::RegisterState::Map &Map) {
+  using namespace abi;
+  using namespace ABIAnalyses;
+  using RegisterState = abi::RegisterState::Values;
+  using Register = model::Register::Values;
+
+  auto GPRArguments = listGeneralPurposeArgumentRegisters(Binary->DefaultABI);
+  auto GPRRetVals = listGeneralPurposeReturnValueRegisters(Binary->DefaultABI);
+  auto VRArguments = listVectorArgumentRegisters(Binary->DefaultABI);
+  auto VRRetVals = listVectorReturnValueRegisters(Binary->DefaultABI);
+
+  auto TryGetRegisterState =
+    [this](Register Reg,
+           RegisterStateMap &ABIRegisterMap) -> std::optional<RegisterState> {
```

Comment:
I feel like this a pretty sad way to format a function, but there's nothing much that can be done to improve it if we want to leave it as a lambda. Maybe it's worth to look into upgrading it into a free (or even a member) function?

Reply:
If the function formatting is not among our upmost concerns, I'd would leave a lambda function, as it seems to be actually the case for it to exist here (it has captures, being used only within this function, not really needed to promote it to member function).

Reply:
Yeah, that's what I meant by "member function", because to me personally a lambda that only captures `this` and is only used within the class is the same thing as a private member - except it looks somewhat worse.

Reply:
Moved it as a member function!